promotion/src/main/java/org/cryptimeleon/incentive/promotion/reward/Reward.java

```diff
@@ -35,19 +52,53 @@
      * @param basketPoints the points that the basket is worth
      * @return and optional vector, which returns satisfying points vector, or empty if none was found
      */
-    Optional<Vector<BigInteger>> computeSatisfyingNewPointsVector(Vector<BigInteger> tokenPoints, Vector<BigInteger> basketPoints);
+    public abstract Optional<Vector<BigInteger>> computeSatisfyingNewPointsVector(Vector<BigInteger> tokenPoints, Vector<BigInteger> basketPoints);
 
-    /**
-     * This function returns a representation of the actual reward.
-     *
-     * @return a representation of the reward.
-     */
-    RewardSideEffect getSideEffect();
+    public UUID getRewardId() {
+        return this.rewardId;
+    }
 
-    /**
-     * Returns a random id that uniquely determines the reward.
-     *
-     * @return the reward id
-     */
-    UUID getRewardId();
+    public String getRewardDescription() {
+        return this.rewardDescription;
+    }
+
+    public RewardSideEffect getRewardSideEffect() {
+        return this.rewardSideEffect;
+    }
+
+    public boolean equals(final Object o) {
+        if (o == this) return true;
+        if (!(o instanceof Reward)) return false;
+        final Reward other = (Reward) o;
+        if (!other.canEqual((Object) this)) return false;
+        final Object this$rewardId = this.getRewardId();
+        final Object other$rewardId = other.getRewardId();
+        if (this$rewardId == null ? other$rewardId != null : !this$rewardId.equals(other$rewardId))
+            return false;
+        final Object this$rewardDescription = this.getRewardDescription();
```

Comment:
what do these $ mean?

Reply:
Good question, I guess this is just nice syntax for this auto-generated method.